src/app/shared/checker-workflow.service.ts

```diff
@@ -29,140 +29,169 @@ import { WorkflowService } from './workflow.service';
-                if ((<Workflow>entry).mode === Workflow.ModeEnum.STUB) {
-                    return true;
-                } else {
-                    return false;
-                }
-            }
-        }));
-        this.entry$.subscribe((entry: Entry) => {
-            this.entry = entry;
-            if (!entry || !entry.checker_id) {
-                this.checkerWorkflow$.next(null);
-                return;
-            }
-            const id = entry.checker_id;
-            const publicPage = this.publicPage;
+  // The checker workflow's path (ex. 'github.com/A/l')
+  public checkerWorkflowPath$: Observable<string>;
+  // The checker workflow if it exists
+  public checkerWorkflow$ = new BehaviorSubject<Workflow>(null);
+  // Whether we're on a public page or not (my-tools/my-workflows)
+  public publicPage$: Observable<boolean>;
+  // The current public page status
+  private publicPage: boolean;
+  // The current checker workflow
+  private checkerWorkflow: Workflow;
+  // The current entry being looked at
+  public entry$: Observable<Entry>;
+  // Whether the current entry has a parent or not
+  public hasParentEntry$: Observable<boolean>;
+  public hasParentEntry: boolean;
+  public entry: Entry;
+  // Whether current entry is a stub or not
+  public isStub$: Observable<boolean>;
+  private ngUnsubscribe: Subject<{}> = new Subject();
+  constructor(private workflowsService: WorkflowsService, private stateService: StateService, private workflowService: WorkflowService,
+    private containerService: ContainerService, private router: Router, private containersService: ContainersService) {
+    this.publicPage$ = this.stateService.publicPage$;
+    this.publicPage$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((publicPage: boolean) => this.publicPage = publicPage);
+    this.entry$ = observableMerge(this.containerService.tool$, this.workflowService.workflow$).pipe(
+      filter(x => x != null),
+      distinctUntilChanged());
+    this.hasParentEntry$ = this.entry$.pipe(map((entry: Entry) => {
+      if ((<Workflow>entry).parent_id) {
+        return true;
+      } else {
+        return false;
+      }
+    }));
+    this.checkerWorkflowPath$ = this.checkerWorkflow$.pipe(map((workflow: Workflow) => {
+      if (workflow) {
+        return workflow.full_workflow_path;
+      } else {
+        return null;
+      }
+    }));
+    this.checkerWorkflow$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((workflow: Workflow) => {
+      this.checkerWorkflow = workflow;
+    });
+    this.isStub$ = this.entry$.pipe(map((entry: Entry) => {
+      if (!this.isEntryAWorkflow(entry)) {
+        return false;
+      } else {
+        if ((<Workflow>entry).mode === Workflow.ModeEnum.STUB) {
+          return true;
+        } else {
+          return false;
+        }
+      }
+    }));
+    this.entry$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((entry: Entry) => {
+      this.entry = entry;
+      if ((<Workflow>entry).parent_id) {
```

Comment:
Simpler with `this.hasParentEntry = ((<Workflow>entry).parent_id)`?

Reply:
compiler error, different type

Reply:
hmm...i'll just change this.hasParentEntry to this.parentId